filament/include/filament/LightManager.h

```diff
@@ -283,6 +314,75 @@ class UTILS_PUBLIC LightManager : public FilamentAPI {
         float maxShadowDistance = 0.3;
     };
 
+    struct ShadowCascades {
+        /**
+         * Utility method to compute ShadowOptions::cascadeSplitPositions according to a uniform
+         * split scheme.
+         *
+         * @warning This API is still experimental and subject to change.
+         *
+         * @param splitPositions    a float array to write the split positions into
+         * @param cascades          the number of shadow cascades
+         */
+        static void computeUniformSplits(float splitPositions[3], uint8_t cascades) {
+            size_t s = 0;
+            for (size_t c = 1; c < cascades; c++) {
+                splitPositions[s++] = (float) c / cascades;
+            }
+        }
+
+        /**
+         * Utility method to compute ShadowOptions::cascadeSplitPositions according to a logarithmic
+         * split scheme.
+         *
+         * @warning This API is still experimental and subject to change.
+         *
+         * @param splitPositions    a float array to write the split positions into
+         * @param cascades          the number of shadow cascades
+         * @param near              the camera near plane
+         * @param far               the camera far plane
+         */
+        static void computeLogSplits(float splitPositions[3], uint8_t cascades,
```

Comment:
Do we really want to always take an array of 3? It could just be a float* with the contract that we will write up to 3 cascades. Note that your implementation doesn't clamp the `cascades` parameter so you can write out of bounds anyway.

Reply:
Actually, I think we shouldn't have any code in public headers -- because you can never change it.

The only code that is OK in public headers is templates (sometimes), and helpers -- i.e. things that the user could do themselves.

Reply:
I had this in the public header because I viewed these functions as "helpers," but the inability to change them + the use of an `std::` function has convinced me to make the implementations private.